webapp/lib/app/configurator/standard_messages/controller_messages_helper.dart

```diff
@@ -9,81 +22,184 @@ class StandardMessagesManager {
+          isMessageEdited = true;
+        }
+
+        // todo: moved across categories
+        // todo: moved across groups
+        // todo: rearranged within groups
+        if (isMessageEdited) {
+          editedMessages.add(localMessage);
+        }
+      }
+    }
+
+    return MessagesDiffData(unsavedCategoryIds, unsavedGroupIds, unsavedMessageIds, editedMessages, deletedMessages);
+  }
 
   /// Returns the list of messages being managed.
-  List<model.SuggestedReply> get standardMessages => categories.values.fold([], (result, category) => result..addAll(category.messages));
+  List<model.SuggestedReply> get standardMessagesInStorage => storageCategories.values.fold([], (result, category) => result..addAll(category.messages));
+  List<model.SuggestedReply> get standardMessagesInLocal => localCategories.values.fold([], (result, category) => result..addAll(category.messages));
 
-  model.SuggestedReply getStandardMessageById(String id) => standardMessages.singleWhere((r) => r.suggestedReplyId == id);
+  model.SuggestedReply getStandardMessageById(String id) => standardMessagesInLocal.singleWhere((r) => r.suggestedReplyId == id);
 
 
-  model.SuggestedReply addStandardMessage(model.SuggestedReply standardMessage) {
-    if (standardMessages.any((element) => element.suggestedReplyId == standardMessage.suggestedReplyId)) {
-      updateStandardMessage(standardMessage);
+  model.SuggestedReply addStandardMessageInLocal(model.SuggestedReply standardMessage) {
+    if (standardMessagesInLocal.any((element) => element.suggestedReplyId == standardMessage.suggestedReplyId)) {
+      updateStandardMessageInLocal(standardMessage);
       return null;
     }
-    categories.putIfAbsent(standardMessage.categoryId, () => new MessageCategory(standardMessage.categoryId, standardMessage.category, standardMessage.categoryIndex));
-    categories[standardMessage.categoryId].groups.putIfAbsent(standardMessage.groupId, () => MessageGroup(standardMessage.groupId, standardMessage.groupName, standardMessage.groupIndexInCategory));
-    categories[standardMessage.categoryId].groups[standardMessage.groupId].messages[standardMessage.suggestedReplyId] = standardMessage;
+    localCategories.putIfAbsent(standardMessage.categoryId, () => new MessageCategory(standardMessage.categoryId, standardMessage.category, standardMessage.categoryIndex));
+    localCategories[standardMessage.categoryId].groups.putIfAbsent(standardMessage.groupId, () => MessageGroup(standardMessage.groupId, standardMessage.groupName, standardMessage.groupIndexInCategory));
+    localCategories[standardMessage.categoryId].groups[standardMessage.groupId].messages[standardMessage.suggestedReplyId] = standardMessage;
     return standardMessage;
   }
 
-  List<model.SuggestedReply> addStandardMessages(List<model.SuggestedReply> standardMessagesToAdd) {
+  model.SuggestedReply onAddStandardMessageFromStorage(model.SuggestedReply standardMessage) {
```

Comment:
nit - let's rename this slightly and make it a private method: `_addStandardMessageInStorage`